src/Core/src/Handlers/SwipeItemMenuItem/SwipeItemMenuItemHandler.iOS.cs

```diff
@@ -44,18 +83,27 @@ public static void MapBackground(ISwipeItemMenuItemHandler handler, ISwipeItemMe
 		public static void MapVisibility(ISwipeItemMenuItemHandler handler, ISwipeItemMenuItem view)
 		{
 			var swipeView = handler.PlatformView.GetParentOfType<MauiSwipeView>();
+
 			if (swipeView != null)
 				swipeView.UpdateIsVisibleSwipeItem(view);
 
 			handler.PlatformView.UpdateVisibility(view.Visibility);
 		}
 
+		void OnSwipeItemFrameChanged(object? sender, EventArgs e)
+		{
+			// Adjust the size of the icon in case of changing the size of the SwipeItem.
+			if (this is ISwipeItemMenuItemHandler swipeItemMenuItemHandler)
+				swipeItemMenuItemHandler.UpdateValue(nameof(ISwipeItemMenuItem.Source));
+		}
```

Comment:
Won't this trigger a re-download of the source? Maybe we should be triggering a layout - or requesting some layout again?

Reply:
@jsuarezruiz please we need feedback on this :)